plenum/server/node.py

```diff
@@ -571,6 +571,7 @@ def primaries(self, ps):
         self._primaries = ps
         for r in self.replicas.values():
             r.set_primaries(ps)
+        self.write_manager.future_primary_handler.set_primaries(self.viewNo, ps)
```

Comment:
Why do we need to call it here?
Should we call it from view change service when changing the primary?

Reply:
we call `self.primaries = ` after catch up, in allLedgersCaughtUp method